front_end/src/backend/downloads.js

```diff
@@ -14,98 +14,70 @@
  * limitations under the License.
  */
 
-// In these functions the 'fileName' parameter letiables refer to filenames of '2mib.txt' '64mib.txt' and '256mib.txt'
-export const REGIONS_MAP = {
-    "northamerica-northeast1":"Montréal",
-    "northamerica-northeast2":"Toronto",
-    "us-central1":"Iowa",
-    "us-east1":"South Carolina",
-    "us-east4":"Northern Virginia",
-    "us-east5":"Columbus",
-    "us-south1":"Dallas",
-    "us-west1":"Oregon",
-    "us-west2":"Los Angeles",
-    "us-west3":"Salt Lake City",
-    "us-west4":"Las Vegas",
-    "southamerica-east1":"São Paulo",
-    "southamerica-west1":"Santiago",
-    "europe-central2":"Warsaw",
-    "europe-north1":"Finland",
-    "europe-southwest1":"Madrid",
-    "europe-west1":"Belgium",
-    "europe-west2":"London",
-    "europe-west3":"Frankfurt",
-    "europe-west4":"Netherlands",
-    "europe-west6":"Zürich",
-    "europe-west8":"Milan",
-    "europe-west9":"Paris",
-    "asia-east1":"Taiwan",
-    "asia-east2":"Hong Kong",
-    "asia-northeast1":"Tokyo",
-    "asia-northeast2":"Osaka",
-    "asia-northeast3":"Seoul",
-    "asia-south1":"Mumbai",
-    "asia-south2":"Delhi",
-    "asia-southeast1":"Singapore",
-    "asia-southeast2":"Jakarta",
-    "australia-southeast1":"Sydney",
-    "australia-southeast2":"Melbourne"
-};
+// In these functions the 'fileName' parameter variables refer to filenames of '2mib.txt' '64mib.txt' and '256mib.txt'
+import { REGIONS_MAP, FILESIZE_BYTES, FILESIZE_MIB } from "./common.js";
+import axios from "axios";
 
+export class Downloads {
+    _builtURL;
 
-async function timeDownload(fileName, bucket) {
-    const axios = require('axios');
-    const bucketName = 'gcsrbpa-' + bucket;
-
-    const URL = `https://storage.googleapis.com/${bucketName}/${fileName}`;
-
-    async function downloadFile() {
-        const start = Date.now();
-        await axios.get(URL)
-        const end = Date.now() - start;
-        return end;
+    async downloadFile(URL) {
+        this._builtURL = URL;
+        try {
+            const start = Date.now();
+            await axios.get(URL);
+            const end = Date.now() - start;
+            return end;
+        } catch (e) {
+            return -1;
+        }
     }
 
-    const timeTaken = await downloadFile();
-    return timeTaken / 1000; // return in units of seconds
-}
+    async timeDownload(fileName, bucket) {
+        if (fileName != "wrongFile" && bucket != "wrongBucket") {
```

Comment:
what's the purpose of this line?

Reply:
I wanted to be able to test the axios request failing (ex bad connection) and the function failing because of other reasons (bad fileName/bucketName) separately. This way I can pass in "wrongFile" and "wrongBucket" and ensure errors aren't thrown in `timeDownload` but that axios still fails in `downloadFile`.

Reply:
I fixed this by stubbing `downloads.downloadFile` with a function that returns -1, then restoring after tests have run. Still a bit quirky (requires stubbing and restoring for every test, even though most don't test this behaviour), but I can take the if statement out of the source code